### PR TITLE
test: implement nodeclaim.spec.terminationGracePeriod CEL tests

### PR DESCRIPTION
**Pull request by wmgroot**
**Description**
Implements CEL tests for the new `terminationGracePeriod` field on NodeClaim.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/v1/nodeclaim.go [hunk 1]
```diff
@@ -59,6 +59,9 @@ type NodeClaimSpec struct {
 	//
 	// The feature can also be used to allow maximum time limits for long-running jobs which can delay node termination with preStop hooks.
 	// If left undefined, the controller will wait indefinitely for pods to be drained.
+	// +kubebuilder:validation:Minimum:=0
```
Comment (njtran) [inline pkg/apis/v1/nodeclaim.go:62]: you shouldn't need the minimum. We just do the validation by regex.

File: pkg/apis/v1beta1/nodeclaim.go [hunk 2]
```diff
@@ -66,6 +66,8 @@ type NodeClaimSpec struct {
 	//
 	// The feature can also be used to allow maximum time limits for long-running jobs which can delay node termination with preStop hooks.
 	// If left undefined, the controller will wait indefinitely for pods to be drained.
+	// +kubebuilder:validation:Pattern=`^([0-9]+(\\.[0-9]+)?(ms|s|m|h|d|w|y))+$`
```
Comment (njtran) [inline pkg/apis/v1beta1/nodeclaim.go:69]: I discussed with @wmgroot, but we intentionally only keep it as s/m/h, due to inconsistencies in how days/weeks/years are calculated.  @cnmcavoy can you add that back in as it was? 

https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1/nodepool.go#L84

Comment (wmgroot) [reply pkg/apis/v1beta1/nodeclaim.go:69]: I'm testing out the branch with day+ removed.

Comment (wmgroot) [reply pkg/apis/v1beta1/nodeclaim.go]: Should be all good now.

File: kwok/charts/crds/karpenter.sh_nodeclaims.yaml [hunk 3]
```diff
@@ -267,6 +267,7 @@ spec:
 
                     The feature can also be used to allow maximum time limits for long-running jobs which can delay node termination with preStop hooks.
                     If left undefined, the controller will wait indefinitely for pods to be drained.
+                  pattern: ^([0-9]+(\\.[0-9]+)?(s|m|h))+$
```
Comment (njtran) [inline kwok/charts/crds/karpenter.sh_nodeclaims.yaml:270]: why do we need the middle bit? 
```suggestion
                  pattern: ^(([0-9]+(s|m|h))+)$
```

**Review by njtran — COMMENTED** [review]
/lgtm
/approve

Thanks for the commit!

**Pull request merged**